src/main/resources/org/jenkinsci/plugins/workflow/steps/durable_task/ShellStep/help-script.html

```diff
@@ -9,4 +9,14 @@
     Otherwise the system default shell will be run, using the <code>-xe</code> flags
     (you can specify <code>set +e</code> and/or <code>set +x</code> to disable those).
     </p>
+    <p>
+    NOTE: The script is launched and monitored using a binary with the prefix <code>durable-task-monitor-</code>.
+          Because the binary is launched as a background process, please ensure that the target script environment
+          is able to reap orphaned processes (i.e. zombie-reaping). For example, in Docker, you may use the
+          <a href="https://docs.docker.com/engine/reference/run/?#specify-an-init-process"><code>--init</code> flag</a>,
+          while in Kubernetes, you may leverage
+          <a href="https://kubernetes.io/docs/tasks/configure-pod-container/share-process-namespace/"> shared process namespace</a>.
+          With shared process namespaces, the pod sandbox will be assigned PID 1 and assume
+          <a href="https://github.com/kubernetes/kubernetes/blob/master/build/pause/pause.c#L37">zombie-reaping responsibilities</a>.
```

Comment:
Maybe that's too specific? I just couldn't find any official documentation on the zombie reaping for some reason. I found PRs and discussion and blog posts, but nothing on kubernetes.io. Maybe i'm searching for the wrong keywords?

Reply:
Yeah, I wouldn't link directly to source code. I think the link to shared process namespaces is good enough. Any idea of how common this kind of setting is, or if there are any alternatives? It looks like it has some drawbacks in terms of security, so it seems less than ideal if that is the only way to get things working in Kubernetes.

Reply:
It's actually enabled by default (as of `v1.15`)

Reply:
The only other way i'm aware of in Kubernetes, would be to then roll your own `init` process for each container

Reply:
If I understand correctly, the ability to use the feature is what is enabled by default, but you have to explicitly use it by adding `shareProcessNamespace: true` to the Pod spec. I think disabling the feature makes it so that `shareProcessNamespace` is not a valid field for Pod specs.

Reply:
Looks like the feature has been discussed for a few years now, but only got introduced as a beta in its current incarnation about a year ago. That's about the only insight I can give about how common of a use case it is.